dnf5/commands/repoquery/repoquery.cpp

```diff
@@ -611,37 +629,20 @@ void RepoqueryCommand::run() {
         full_package_query &= get_installonly_query(ctx.base);
     }
 
-    if (pkg_specs.empty()) {
-        result_pset |= full_package_query;
-    } else {
-        for (const auto & pkg : cmdline_packages) {
-            if (full_package_query.contains(pkg)) {
-                result_pset.add(pkg);
-            }
-        }
-
-        const libdnf::ResolveSpecSettings settings{.ignore_case = true, .with_provides = false};
-        for (const auto & spec : pkg_specs) {
-            libdnf::rpm::PackageQuery package_query(full_package_query);
-            package_query.resolve_pkg_spec(spec, settings, true);
-            result_pset |= package_query;
-        }
-    }
 
     if (querytags_option->get_value()) {
         libdnf::cli::output::print_available_pkg_attrs(stdout);
     } else if (changelogs->get_value()) {
-        libdnf::rpm::PackageQuery wrapped(result_pset);
-        libdnf::cli::output::print_changelogs(wrapped, full_package_query, false, 0, 0);
+        libdnf::cli::output::print_changelogs(full_package_query, full_package_query, false, 0, 0);
```

Comment:
I was confused about this, because the second argument for `print_changelogs` changed during the move, but it doesn't have any effect... However, the query is then unnecessarily  copied and filtered, so wouldn't it be better to pass an empty query or change it in the `print_changelogs`? Neither is necessary in this PR, though (it's actually smaller now than it was before).

Reply:
Yes, I agree. I tried to improve the `print_changelogs` in a new commit. Please take a look.

Reply:
Thanks. FYI, I encountered an unrelated issue with the changelog command (https://github.com/rpm-software-management/dnf5/issues/627) when testing it. Just mentioning if you run into the same.